plugins/processors/cumulative_sum/cumulative_sum.go

```diff
@@ -0,0 +1,104 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package cumulative_sum
+
+import (
+	_ "embed"
+	"fmt"
+	"maps"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type CumulativeSum struct {
+	Fields         []string        `toml:"fields"`
+	ExpiryInterval config.Duration `toml:"expiry_interval"`
+	Log            telegraf.Logger `toml:"-"`
+	accept         filter.Filter
+	cache          map[uint64]*entry
+}
+
+type entry struct {
+	sums map[string]float64
+	seen time.Time
+}
+
+func (*CumulativeSum) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *CumulativeSum) Init() error {
+	if len(c.Fields) == 0 {
+		c.Fields = []string{"*"}
+	}
+	f, err := filter.Compile(c.Fields)
+	if err != nil {
+		return fmt.Errorf("failed to create new field filter: %w", err)
+	}
+	c.accept = f
+
+	c.cache = make(map[uint64]*entry)
+
+	return nil
+}
+
+func (c *CumulativeSum) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	now := time.Now()
+
+	// Cleanup cache entries that are too old
+	if c.ExpiryInterval > 0 {
+		threshold := now.Add(-time.Duration(c.ExpiryInterval))
+		maps.DeleteFunc(c.cache, func(_ uint64, e *entry) bool {
+			return e.seen.Before(threshold)
+		})
+	}
```

Comment:
I think this should maybe go to the end of the function to allow updates in the current cycle to be taken into account... What do you think?

Reply:
Initially, having this at the beginning of the function made sense when we started it not oftener that ResetInterval. Let's move it to the end so that it is consistent with other plugins.